src/legacy/core_plugins/metrics/index.ts

```diff
@@ -45,7 +45,9 @@ export default function(kibana: any) {
 
     init(server: Legacy.Server) {
       const initializerContext = {} as PluginInitializerContext;
-      const core = { http: { server } } as CoreSetup | any; // TODO: Remove 'any' as soon as CoreSetup is completed.
+      // TODO: Remove '@ts-ignore' as soon as CoreSetup is completed.
+      // @ts-ignore
+      const core = { http: { server } } as CoreSetup;
```

Comment:
I see the issue you are running into here -- since we are manually constructing pieces of the `CoreSetup` interface, it makes it a little tricky to maintain an interface like this.

IMO it would still be preferable to use a workaround that doesn't require completely `@ts-ignore`ing the line... I think any of the following options would work as an interim solution, and at least would let you avoid that `ignore`:
```typescript
const core = { http: { server } as unknown } as CoreSetup;
// or
const core = { http: { server } } as Partial<CoreSetup['http']>;
```

Alternatively, there's probably some way we could construct our own interfaces for the shims, though TBH I haven't spent much time yet thinking about how conventions for that would look, but would be open to any thoughts you have!

Reply:
From point of view it may look reasonable to create a temporary stub like:
```
export interface CustomCoreSetup {
  http: {
    server: any;
  };
}
```
And then merge two interfaces: the permanent and the temporary ones.
```
const core = { http: { server } } as CoreSetup & CustomCoreSetup;
```

Done as supposed. I hope you like the idea